content/providers/migration.md

```diff
@@ -19,7 +19,165 @@ Documentation coming soon.
 For manual migration to the family Official Providers refer to the
 [migration guide]({{<ref "/knowledge-base/migrate-to-provider-families">}}).
 
-Documentation for the automated tooling coming soon.
+`family-migrator` is a tooling for the automated migration process. The tool
+supports interactive and non-interactive migration to the family Official
+Providers. The instructions on how to use tooling are below.
+
+The `family-migrator` tool has two sub-commands, `generate` and `execute`.
+
+<!-- vale alex.ProfanityUnlikely = NO -->
+<!-- ignore "executed" -->
+- `generate`: For generating the migration plan. After the tool creates the
+migration plan, asks if the user wants to continue to perform the created plan.
+
+- `execute`: For only performing the generated plan. To run this command,
+you must have a generated plan. Users can use this option for a generated but
+un-executed plan.
+<!-- vale alex.ProfanityUnlikely = YES -->
+
+The `family-migrator` tool needs several inputs for generating and executing the
+plan. It prompts all required options to the user after it runs.
+
+### Collecting inputs
+
+After running the `family-migrator generate`, tool asks a series of questions
+to user:
+
+1. `? Please specify the path for the migration plan`: The answer represents the
+output path of migration plan. Users need to provide a path for generating the
+plan and patch files.
+2. `? Please provide the registry and organization for the provider family
+packages`: The user specifies the family provider registry and organization to
+migrate. The format must be `<registry host>/<organization>`,
+like: _xpkg.upbound.io/upbound_.
+3. `? Please select the providers that will be migrated`:
+The tool needs the versions of the providers that it wants to migrate. For this
+reason, the user must first select the monolithic providers she wishes to
+migrate. The tool supports multiple selections.
+4. `? Please specify the version of the provider-xxx family`: The user must
+provide the family versions for every selected provider.
+If you want to migrate to the newest version, please see the [Upbound
+Marketplace](https://marketplace.upbound.io/) for the latest version.
+5. `? Please enter the URL of the migration source Configuration package`: The
+user must provide the URL of the currently installed Configuration Package. The
+user can take the value from the `spec.package` path of
+`Configuration.pkg.crossplane.io/v1` resource.
+6. `? Please enter the URL of the migration target Configuration package`: The
+tool builds a new configuration package during migration. Specify the URL to
+which the user pushes the built package.
```

Comment:
If a configuration package exists in the cluster, the newly generated package must be built and pushed. Because this new package URL will now be referenced in the configuration package in the cluster. Therefore, the configuration package resource will only be ready if the newly built package is pushed.

However, this step may not seem necessary because the cluster has no configuration package resource. On the other hand, the user can skip answering this and Skip the push step while running Step-by-Step.

Reply:
That's fair enough. Maybe we can add a hint in the docs to allude to this?